dash/SQLConnection.py

```diff
@@ -19,9 +22,34 @@ def query(self, query:str) -> pd.DataFrame:
 			pd.DataFrame: result of the query
 		"""
 		try:
+			lock.acquire(True)
 			res = pd.read_sql_query(query, self.conn)
 			print(f"Querying: '{query}'")
 			return res
 		except Exception as e:
 			print(e)
-		print('\n')
\ No newline at end of file
+			print('\n')
+		finally:
+			lock.release()
+
+	def execute(self, query:str) -> list:
+		"""
+		Execute a query on the db
+
+		Args:
+				query (str): query string
+
+		Returns:
+				list: returned from query
+		"""
+		try:
+			lock.acquire(True)
+			res = self.curs.execute(query)
+			res = res.fetchall()
+			print(f"'{query}' executed")
+			return res
+		except Exception as e:
+			print(e)
+			print('\n')
+		finally:
+			lock.release()
```

Comment:
`query` and `execute` both necessary but maybe change their names to be more descriptive? `query` returns a df and `execute` returns a list so maybe call them `q_df` and `q_list` or something like that

Reply:
I like that, distinguish their difference in their names

Reply:
changed to get_df and get_list